ONSdigital/sdc-global-design-patterns
# Add Typeahead Component

### What is the context of this PR?
Added a typeahead component that should hopefully address the accessibility issues raised in the DAC report.

This should address:
* DAC_TECH_Forms_01
* DAC-USER-SRVO-T1-01
* DAC-USER-SRVO-T2-01
* DAC_TECH_Retest_Aria01

### How to review 
All new code is under `components/02-form-elements/typeahead`

Run `yarn start`

Standard country picker example (country list might not be 100% as they're generated from a library):
http://localhost:3000/components/detail/typeahead--country-typeahead

Welsh version of country picker (demonstrates alternative values functionality):
http://localhost:3000/components/detail/typeahead--country-typeahead-cy

* Check keyboard navigation
* Check VoiceOver
* Check label is transferred to input and is read on focus
* Check instructions make sense for touch device users also
* Check ability to select results after software keyboard is dismissed
